src/data/entities/D2ExpressionParser.ts

```diff
@@ -0,0 +1,114 @@
+import * as xp from "@dhis2/expression-parser";
+import _c from "../../domain/entities/generic/Collection";
+import { Either } from "../../domain/entities/generic/Either";
+
+export type ProgramRuleVariableType = "text" | "number" | "date" | "boolean";
+
+export type ProgramRuleVariableName = string;
+export type ProgramRuleVariableValue = {
+    type: ProgramRuleVariableType;
+    value: string;
+};
+
+const VariableValueTypeMap: Record<ProgramRuleVariableType, xp.ValueType> = {
+    text: xp.ValueType.STRING,
+    boolean: xp.ValueType.BOOLEAN,
+    date: xp.ValueType.DATE,
+    number: xp.ValueType.NUMBER,
+};
+
+export class D2ExpressionParser {
+    public evaluateRuleEngineCondition(
+        ruleCondtion: string,
```

Comment:
Add some docs somewhere (probably the easiest is a link to DHIS2 documentation) so we know what kind of conditions we can do.